Feature/simple auth with cookie

iryabov opened the pull request.

iryabov committed:
feat: add jwt cookie in the sign-in response

RomanDavlyatshin commented on admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt:
```diff
@@ -91,6 +92,7 @@ fun Route.signInRoute() {
         val userView = authService.signIn(loginPayload)
         val token = tokenService.issueToken(userView)
         call.response.header(HttpHeaders.Authorization, token)
```
Passing token in _both_ header and http-only cookie defeats the purpose of httpOnly cookie mechanism - to make token inaccessible via javascript in the webbrowser.
Lets discuss if we can keep it _only_ in cookie. I don't see much difference in extracting value from header of from cookie in terms of usability

iryabov committed:
feat: add routes, services and models for api keys

EPMDJ-10730

iryabov committed:
refactor: move user api keys to UserApiKeyRoutes.kt file
tests: add tests for UserApiKeyRoutes.kt file

EPMDJ-10730

iryabov committed:
Merge branch 'feature/api-keys-EPMDJ-10722' into feature/create-api-for-api-keys-EPMDJ-10730

# Conflicts:
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt

iryabov committed:
docs: add kdoc for Api Key routes

EPMDJ-10730

iryabov committed:
refactor: rename expired to expiresAt, created to createdAt

EPMDJ-10730

iryabov committed:
feat: add database api-key repository implementation

EPMDJ-10739

iryabov committed:
feat: add database api-key repository implementation

EPMDJ-10739

iryabov committed:
feat: add implementation of api key service
tests: add tests for api key service, user keys routes, management keys routes

EPMDJ-10739

iryabov committed:
tests: make insertTestUser function private

EPMDJ-10739

iryabov committed:
tests: move postgres database creation to a separate class

EPMDJ-10739

iryabov committed:
Merge pull request #356 from Drill4J/feature/create-api-for-api-keys-EPMDJ-10730

Feature/create api for api keys epmdj 10730

iryabov committed:
Merge remote-tracking branch 'origin/feature/api-keys-EPMDJ-10722' into feature/api-keys-services-implementation-EPMDJ-10739

iryabov committed:
tests: add a test for api key expiration date

EPMDJ-10739

iryabov committed:
feat: create a Ktor apiKey authentication provider

EPMDJ-10740

iryabov committed:
refactor: move passwordGenerator and passwordValidator to separate interface classes

EPMDJ-10740

iryabov committed:
feat: add sign-in through api key function

EPMDJ-10740

iryabov committed:
feat: add api key configuration module
tests: add tests for sign-in through api key function and api key module

EPMDJ-10740

iryabov committed:
fix: fix the API key generation response, return the key itself instead of a key hash

EPMDJ-10740

iryabov committed:
chore: add licences

EPMDJ-10740

iryabov committed:
feat: add api-key authentication method to Drill4J Admin app

EPMDJ-10740

iryabov committed:
chore: add licences

EPMDJ-10740

iryabov committed:
feat: add the api key identifier to the api key string, to look up a specific key in the database
feat: add apiKeyBuilder interface and simple implementation

EPMDJ-10740

iryabov committed:
fix: change jwt secret byte array encoding from toString to base64

EPMDJ-10740

iryabov committed:
chore: add licences

EPMDJ-10740

iryabov committed:
feat: add setting the length of the secret part of the API key

EPMDJ-10740

iryabov committed:
feat: add setting the length of the secret part of the API key

EPMDJ-10740

iryabov committed:
feat: add the ability to enable and disable various types of authentication through the configuration properties:
`DRILL_OAUTH2_ENABLED`
`DRILL_SIMPLE_AUTH_ENABLED`

feat: remove the configuration property `DRILL_AUTH_TYPE`

EPMDJ-10742

iryabov committed:
refactor: remove the auth type configuration property

EPMDJ-10742

iryabov committed:
feat: add property `enabled` to the Oauth2Config class and SimpleAuthCOnfig file

EPMDJ-10742

iryabov committed:
refactor: move jwt configuration to a separate file

EPMDJ-10742

iryabov committed:
fix: return OAuth2Config di bind

EPMDJ-10742

iryabov committed:
feat: remove deprecated /login route

EPMDJ-10745

iryabov committed:
feat: remove basic authentication because there is API keys

EPMDJ-10743

iryabov committed:
feat: remove EnvUserRepository

EPMDJ-10744

iryabov committed:
feat: change the root path to configure simple authentication, now it is `drill.auth.simple`
feat: change the root path to configure password strength, now it is `drill.auth.simple.passwordStrength`

EPMDJ-10746

iryabov committed:
refactor: rename Models to UserModels

EPMDJ-10746

iryabov committed:
refactor: move auth config models to the separate file AuthConfigModels.kt

EPMDJ-10746

iryabov committed:
feat: add a check for deleting an API key issued by someone other than the current user

EPMDJ-10739

iryabov committed:
tests: add asserts

EPMDJ-10739

iryabov committed:
refactor: change NullPointerException to IllegalStateException

EPMDJ-10739

iryabov committed:
Merge branch 'feature/api-keys-services-implementation-EPMDJ-10739' into feature/api-key-authentication-method-implementation-EPMDJ-10740

# Conflicts:
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/repository/ApiKeyRepository.kt
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/service/impl/ApiKeyServiceImpl.kt

iryabov committed:
refactor: merge with feature/api-keys-services-implementation-EPMDJ-10739 branch

EPMDJ-10739

iryabov committed:
tests: add test for custom api key header

EPMDJ-10739

iryabov committed:
refactor: improve code style

EPMDJ-10739

iryabov committed:
tests: rename test cases

EPMDJ-10740

iryabov committed:
refactor: add checks for api key format

EPMDJ-10740

iryabov committed:
refactor: rename variable for better readability

EPMDJ-10740

iryabov committed:
Merge pull request #359 from Drill4J/feature/api-key-authentication-method-implementation-EPMDJ-10740

Feature/api key authentication method implementation epmdj 10740

iryabov committed:
Merge remote-tracking branch 'origin/feature/api-keys-EPMDJ-10722' into feature/auth-modules-refactoring-EPMDJ-10742

iryabov committed:
refactor: rename authStatusPages function
feat: add handling of user and api-key not found exceptions

EPMDJ-10742

iryabov committed:
Merge pull request #360 from Drill4J/feature/auth-modules-refactoring-EPMDJ-10742

Feature/auth modules refactoring epmdj 10742

iryabov committed:
Merge remote-tracking branch 'origin/feature/api-keys-EPMDJ-10722' into feature/remove-login-route-EPMDJ-10745

iryabov committed:
Merge pull request #361 from Drill4J/feature/remove-login-route-EPMDJ-10745

Feature/remove login route epmdj 10745

iryabov committed:
Merge pull request #362 from Drill4J/feature/remove-basic-auth-EPMDJ-10743

feat: remove basic authentication because there is API keys

iryabov committed:
Merge pull request #363 from Drill4J/feature/remove-users-from-env-EPMDJ-10744

feat: remove EnvUserRepository

iryabov committed:
Merge remote-tracking branch 'origin/feature/api-keys-EPMDJ-10722' into feature/api-key-cleanup-EPMDJ-10746

iryabov committed:
fix: fix kdoc

EPMDJ-10746

iryabov committed:
docs: add kdoc for ApiKeyRepository

EPMDJ-10746

iryabov committed:
docs: add kdoc for ApiKeyService

EPMDJ-10746

iryabov committed:
docs: add kdoc for ApiKeyBuilder.kt file

EPMDJ-10746

iryabov committed:
docs: add kdoc for OAuthService interface

EPMDJ-10746

iryabov committed:
docs: add kdoc for OAuthMapper interface

EPMDJ-10746

iryabov committed:
docs: add kdoc for SecretGenerator interface

EPMDJ-10746

iryabov committed:
feat: rename the env variable DRILL_SIMPLE_AUTH_ENABLED to DRILL_AUTH_SIMPLE_ENABLED

EPMDJ-10746

iryabov committed:
Merge pull request #364 from Drill4J/feature/api-key-cleanup-EPMDJ-10746

Feature/api key cleanup epmdj 10746

iryabov committed:
Merge remote-tracking branch 'origin/feature/oauth-support-EPMDJ-10702' into feature/api-keys-EPMDJ-10722

# Conflicts:
#	admin-core/src/main/kotlin/com/epam/drill/admin/endpoints/admin/AgentEndpoints.kt
#	admin-core/src/main/kotlin/com/epam/drill/admin/endpoints/admin/DrillAdminEndpoints.kt
#	admin-core/src/main/kotlin/com/epam/drill/admin/endpoints/plugin/PluginDispatcher.kt
#	admin-core/src/main/kotlin/com/epam/drill/admin/group/Handler.kt

iryabov committed:
refactor: merge with feature/oauth-support-EPMDJ-10702 branch

EPMDJ-10722

iryabov committed:
Merge remote-tracking branch 'origin/feature/oauth-support-EPMDJ-10702' into feature/api-keys-EPMDJ-10722

# Conflicts:
#	admin-core/src/main/resources/application.conf

iryabov committed:
feat: make oauth2 disabled by default

EPMDJ-10722

bodyangug committed:
feat: add caffeine dependency

bodyangug committed:
feat: add impl of caching

iryabov committed:
refactor: remove unused function

EPMDJ-10722

bodyangug committed:
feat: remove old impl of cache

bodyangug committed:
feat: downgrade caffeineVersion for java 8

bodyangug committed:
feat: move cache logic to CacheService

bodyangug committed:
feat: add new cache logic to validate

bodyangug committed:
feat: add apiKeys to ApiKeyConfig

bodyangug committed:
feat: add tests

bodyangug committed:
refactor: change hour to minutes for api key expiration

bodyangug committed:
refactor: code style change

bodyangug committed:
refactor: rename CacheService to ApiKeyCacheService

bodyangug committed:
refactor: rename ApiKeyCacheServiceImpl to CoffeineCacheService

bodyangug committed:
refactor: switch org.junit.Assert.assertEquals to kotlin.test.assertEquals

bodyangug committed:
refactor: change visibility cache val to private

bodyangug committed:
refactor: rework tests

bodyangug committed:
refactor: add test-case with nullable principle

bodyangug committed:
refactor: add license

bodyangug committed:
Merge branch 'feature/api-keys-EPMDJ-10722' of https://github.com/Drill4J/admin into feature/cache-api-keys-EPMDJ-10750

bodyangug committed:
refactor: add license

bodyangug committed:
refactor: add license

bodyangug committed:
refactor: rework tests

bodyangug committed:
feat: change IllegalArgumentException to NotAuthenticatedException

bodyangug committed:
tests: add test-cases for invalid api-key value

bodyangug committed:
Merge branch 'feature/api-keys-EPMDJ-10722' of https://github.com/Drill4J/admin into feature/incorrect-error-code-for-api-key-EPMDJ-10761

bodyangug committed:
refactor: change test-name & change tested value

iryabov committed:
Merge pull request #370 from Drill4J/feature/incorrect-error-code-for-api-key-EPMDJ-10761

Feature/incorrect error code for api key epmdj 10761

bodyangug committed:
refactor: remove tests & test-framework modules

bodyangug committed:
Merge branch 'feature/oauth-support-EPMDJ-10702' of https://github.com/Drill4J/admin into feature/simple-auth-with-cookie

# Conflicts:
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/config/OAuthModule.kt
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt

bodyangug committed:
refactor: remove double declaration of JWT cookie variable

bodyangug committed:
feat: reconfigured auth to use cookies, eliminating the need for header operations

bodyangug committed:
test: reworked test to set up cookie instead of header

bodyangug committed:
refactor: poor grammar changes

bodyangug committed:
refactor: rename CacheService

iryabov committed:
refactor: add a special exception for invalid API key format

EPMDJ-10722

iryabov committed:
refactor: delete a duplicated classes

EPMDJ-10722

iryabov committed:
Merge branch 'feature/api-keys-EPMDJ-10722' into feature/cache-api-keys-EPMDJ-10750

RomanDavlyatshin committed:
Merge pull request #369 from Drill4J/feature/cache-api-keys-EPMDJ-10750

Feature/cache api keys epmdj 10750

iryabov committed:
Merge remote-tracking branch 'origin/feature/oauth-support-EPMDJ-10702' into feature/api-keys-EPMDJ-10722

# Conflicts:
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt
#	admin-core/src/main/kotlin/com/epam/drill/admin/DrillApplication.kt

iryabov committed:
chore: add licence

EPMDJ-10722

iryabov committed:
Merge remote-tracking branch 'origin/feature/api-keys-EPMDJ-10722' into feature/simple-auth-with-cookie

# Conflicts:
#	admin-auth/src/main/kotlin/com/epam/drill/admin/auth/config/SimpleAuthModule.kt

iryabov committed:
refactor: resolve merge conflicts

EPMDJ-10722
